src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -3079,7 +3079,7 @@ public override BoundNode VisitSequencePoint(BoundSequencePoint node)
 
         public override BoundNode VisitSequencePointExpression(BoundSequencePointExpression node)
         {
-            VisitRvalue(node.Expression);
+            Visit(node.Expression);
```

Comment:
>Visit(node.Expression); [](http://example.com/codeflow?start=12&length=23)

Are both modified code paths covered by the added test?

Reply:
Yes. Below is the bound tree for the method, annotated to illustrate where those two kinds of nodes show up:

```
block
├─hasUnsafeModifier: False
├─instrumentation
└─statements
  ├─sequencePointWithSpan
  │ ├─statementOpt
  │ └─span: [84..85)
  ├─sequencePoint
  │ └─statementOpt
  │   └─returnStatement
  │     ├─refKind: None
  │     ├─expressionOpt
  │     └─@checked: False
  └─tryStatement
    ├─tryBlock
    │ └─block
    │   ├─hasUnsafeModifier: False
    │   ├─instrumentation
    │   └─statements
    │     ├─sequencePointWithSpan
    │     │ ├─statementOpt
    │     │ └─span: [127..128)
    │     ├─sequencePoint
    │     │ └─statementOpt
    │     │   └─expressionStatement
    │     │     └─expression
    │     │       └─call
    │     │         ├─receiverOpt
    │     │         │ └─typeExpression
    │     │         │   ├─aliasOpt
    │     │         │   ├─boundContainingTypeOpt
    │     │         │   ├─boundDimensionsOpt
    │     │         │   ├─typeWithAnnotations: System.Console
    │     │         │   └─type: System.Console
    │     │         ├─initialBindingReceiverIsSubjectToCloning: Unknown
    │     │         ├─method: System.Console.WriteLine(string)
    │     │         ├─arguments
    │     │         │ └─fieldAccess
    │     │         │   ├─receiverOpt
    │     │         │   │ └─typeExpression
    │     │         │   │   ├─aliasOpt
    │     │         │   │   ├─boundContainingTypeOpt
    │     │         │   │   ├─boundDimensionsOpt
    │     │         │   │   ├─typeWithAnnotations: string
    │     │         │   │   └─type: string
    │     │         │   ├─fieldSymbol: string.Empty
    │     │         │   ├─constantValueOpt
    │     │         │   ├─resultKind: Viable
    │     │         │   ├─isByValue: False
    │     │         │   ├─isDeclaration: False
    │     │         │   └─type: string
    │     │         ├─argumentNamesOpt: (null)
    │     │         ├─argumentRefKindsOpt: (null)
    │     │         ├─isDelegateCall: False
    │     │         ├─expanded: False
    │     │         ├─invokedAsExtensionMethod: False
    │     │         ├─argsToParamsOpt: (null)
    │     │         ├─defaultArguments: Microsoft.CodeAnalysis.BitVector
    │     │         ├─resultKind: Viable
    │     │         ├─originalMethodsOpt: (null)
    │     │         └─type: void
    │     └─sequencePointWithSpan
    │       ├─statementOpt
    │       └─span: [184..185)
    ├─catchBlocks
    │ └─catchBlock
    │   ├─locals: {e1, e2}
    │   ├─exceptionSourceOpt
    │   │ └─local
    │   │   ├─localSymbol: e1
    │   │   ├─declarationKind: None
    │   │   ├─constantValueOpt
    │   │   ├─isNullableUnknown: False
    │   │   └─type: System.Exception
    │   ├─exceptionTypeOpt: System.Exception
    │   ├─exceptionFilterPrologueOpt
    │   ├─exceptionFilterOpt
    │   │ └─sequencePointExpression // <--------- here
    │   │   ├─expression
    │   │   │ └─sequence // <--------- here
    │   │   │   ├─locals: {System.Exception LoweringTemp.1}
    │   │   │   ├─sideEffects
    │   │   │   ├─value
    │   │   │   │ └─binaryOperator // <--------- returns with a split state (e2 was definitely assigned in one branch, but not in the other)
    │   │   │   │   ├─operatorKind: LogicalBoolAnd
    │   │   │   │   ├─data
    │   │   │   │   ├─resultKind: Viable
    │   │   │   │   ├─left
    │   │   │   │   │ └─sequence
    │   │   │   │   │   ├─sideEffects
    │   │   │   │   │   │ └─assignmentOperator
    │   │   │   │   │   │   ├─left
    │   │   │   │   │   │   │ └─local
    │   │   │   │   │   │   │   ├─localSymbol: System.Exception LoweringTemp.1
    │   │   │   │   │   │   │   ├─declarationKind: None
    │   │   │   │   │   │   │   ├─constantValueOpt
    │   │   │   │   │   │   │   ├─isNullableUnknown: False
    │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   ├─right
    │   │   │   │   │   │   │ └─call
    │   │   │   │   │   │   │   ├─receiverOpt
    │   │   │   │   │   │   │   │ └─local
    │   │   │   │   │   │   │   │   ├─localSymbol: e1
    │   │   │   │   │   │   │   │   ├─declarationKind: None
    │   │   │   │   │   │   │   │   ├─constantValueOpt
    │   │   │   │   │   │   │   │   ├─isNullableUnknown: False
    │   │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   │   ├─initialBindingReceiverIsSubjectToCloning: Unknown
    │   │   │   │   │   │   │   ├─method: System.Exception.InnerException.get
    │   │   │   │   │   │   │   ├─arguments
    │   │   │   │   │   │   │   ├─argumentNamesOpt: (null)
    │   │   │   │   │   │   │   ├─argumentRefKindsOpt: (null)
    │   │   │   │   │   │   │   ├─isDelegateCall: False
    │   │   │   │   │   │   │   ├─expanded: False
    │   │   │   │   │   │   │   ├─invokedAsExtensionMethod: False
    │   │   │   │   │   │   │   ├─argsToParamsOpt: (null)
    │   │   │   │   │   │   │   ├─defaultArguments: Microsoft.CodeAnalysis.BitVector
    │   │   │   │   │   │   │   ├─resultKind: Viable
    │   │   │   │   │   │   │   ├─originalMethodsOpt: (null)
    │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   ├─value
    │   │   │   │   │   │ └─binaryOperator
    │   │   │   │   │   │   ├─operatorKind: NotEqual
    │   │   │   │   │   │   ├─data
    │   │   │   │   │   │   ├─resultKind: Viable
    │   │   │   │   │   │   ├─left
    │   │   │   │   │   │   │ └─local
    │   │   │   │   │   │   │   ├─localSymbol: System.Exception LoweringTemp.1
    │   │   │   │   │   │   │   ├─declarationKind: None
    │   │   │   │   │   │   │   ├─constantValueOpt
    │   │   │   │   │   │   │   ├─isNullableUnknown: False
    │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   ├─right
    │   │   │   │   │   │   │ └─literal
    │   │   │   │   │   │   │   ├─constantValueOpt: ConstantValueNull(null: Nothing)
    │   │   │   │   │   │   │   └─type: object
    │   │   │   │   │   │   └─type: bool
    │   │   │   │   │   └─type: bool
    │   │   │   │   ├─right
    │   │   │   │   │ └─sequence
    │   │   │   │   │   ├─sideEffects
    │   │   │   │   │   │ └─assignmentOperator // <----- assignment to e2
    │   │   │   │   │   │   ├─left
    │   │   │   │   │   │   │ └─local
    │   │   │   │   │   │   │   ├─localSymbol: e2
    │   │   │   │   │   │   │   ├─declarationKind: WithInferredType
    │   │   │   │   │   │   │   ├─constantValueOpt
    │   │   │   │   │   │   │   ├─isNullableUnknown: False
    │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   ├─right
    │   │   │   │   │   │   │ └─call
    │   │   │   │   │   │   │   ├─receiverOpt
    │   │   │   │   │   │   │   │ └─local
    │   │   │   │   │   │   │   │   ├─localSymbol: System.Exception LoweringTemp.1
    │   │   │   │   │   │   │   │   ├─declarationKind: None
    │   │   │   │   │   │   │   │   ├─constantValueOpt
    │   │   │   │   │   │   │   │   ├─isNullableUnknown: False
    │   │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   │   ├─initialBindingReceiverIsSubjectToCloning: Unknown
    │   │   │   │   │   │   │   ├─method: System.Exception.InnerException.get
    │   │   │   │   │   │   │   ├─arguments
    │   │   │   │   │   │   │   ├─argumentNamesOpt: (null)
    │   │   │   │   │   │   │   ├─argumentRefKindsOpt: (null)
    │   │   │   │   │   │   │   ├─isDelegateCall: False
    │   │   │   │   │   │   │   ├─expanded: False
    │   │   │   │   │   │   │   ├─invokedAsExtensionMethod: False
    │   │   │   │   │   │   │   ├─argsToParamsOpt: (null)
    │   │   │   │   │   │   │   ├─defaultArguments: Microsoft.CodeAnalysis.BitVector
    │   │   │   │   │   │   │   ├─resultKind: Viable
    │   │   │   │   │   │   │   ├─originalMethodsOpt: (null)
    │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   ├─value
    │   │   │   │   │   │ └─binaryOperator
    │   │   │   │   │   │   ├─operatorKind: NotEqual
    │   │   │   │   │   │   ├─data
    │   │   │   │   │   │   ├─resultKind: Viable
    │   │   │   │   │   │   ├─left
    │   │   │   │   │   │   │ └─local
    │   │   │   │   │   │   │   ├─localSymbol: e2
    │   │   │   │   │   │   │   ├─declarationKind: WithInferredType
    │   │   │   │   │   │   │   ├─constantValueOpt
    │   │   │   │   │   │   │   ├─isNullableUnknown: False
    │   │   │   │   │   │   │   └─type: System.Exception
    │   │   │   │   │   │   ├─right
    │   │   │   │   │   │   │ └─literal
    │   │   │   │   │   │   │   ├─constantValueOpt: ConstantValueNull(null: Nothing)
    │   │   │   │   │   │   │   └─type: object
    │   │   │   │   │   │   └─type: bool
    │   │   │   │   │   └─type: bool
    │   │   │   │   └─type: bool
    │   │   │   └─type: bool
    │   │   └─type: bool
    │   ├─body
    │   │ └─block
    │   │   ├─hasUnsafeModifier: False
    │   │   ├─instrumentation
    │   │   └─statements
    │   │     ├─sequencePointWithSpan
    │   │     │ ├─statementOpt
    │   │     │ └─span: [290..291)
    │   │     ├─sequencePoint
    │   │     │ └─statementOpt
    │   │     │   └─expressionStatement
    │   │     │     └─expression
    │   │     │       └─call
    │   │     │         ├─receiverOpt
    │   │     │         │ └─typeExpression
    │   │     │         │   ├─aliasOpt
    │   │     │         │   ├─boundContainingTypeOpt
    │   │     │         │   ├─boundDimensionsOpt
    │   │     │         │   ├─typeWithAnnotations: System.Console
    │   │     │         │   └─type: System.Console
    │   │     │         ├─initialBindingReceiverIsSubjectToCloning: Unknown
    │   │     │         ├─method: System.Console.WriteLine(string)
    │   │     │         ├─arguments
    │   │     │         │ └─call
    │   │     │         │   ├─receiverOpt
    │   │     │         │   │ └─local
    │   │     │         │   │   ├─localSymbol: e2
    │   │     │         │   │   ├─declarationKind: None
    │   │     │         │   │   ├─constantValueOpt
    │   │     │         │   │   ├─isNullableUnknown: False
    │   │     │         │   │   └─type: System.Exception
    │   │     │         │   ├─initialBindingReceiverIsSubjectToCloning: Unknown
    │   │     │         │   ├─method: System.Exception.Message.get
    │   │     │         │   ├─arguments
    │   │     │         │   ├─argumentNamesOpt: (null)
    │   │     │         │   ├─argumentRefKindsOpt: (null)
    │   │     │         │   ├─isDelegateCall: False
    │   │     │         │   ├─expanded: False
    │   │     │         │   ├─invokedAsExtensionMethod: False
    │   │     │         │   ├─argsToParamsOpt: (null)
    │   │     │         │   ├─defaultArguments: Microsoft.CodeAnalysis.BitVector
    │   │     │         │   ├─resultKind: Viable
    │   │     │         │   ├─originalMethodsOpt: (null)
    │   │     │         │   └─type: string
    │   │     │         ├─argumentNamesOpt: (null)
    │   │     │         ├─argumentRefKindsOpt: (null)
    │   │     │         ├─isDelegateCall: False
    │   │     │         ├─expanded: False
    │   │     │         ├─invokedAsExtensionMethod: False
    │   │     │         ├─argsToParamsOpt: (null)
    │   │     │         ├─defaultArguments: Microsoft.CodeAnalysis.BitVector
    │   │     │         ├─resultKind: Viable
    │   │     │         ├─originalMethodsOpt: (null)
    │   │     │         └─type: void
    │   │     └─sequencePointWithSpan
    │   │       ├─statementOpt
    │   │       └─span: [345..346)
    │   └─isSynthesizedAsyncCatchAll: False
    ├─finallyBlockOpt
    ├─finallyLabelOpt
    └─preferFaultHandler: False
```